plugins/main/public/components/settings/settings.tsx

```diff
@@ -338,22 +368,19 @@ export class SettingsController {
 
       // Test the connection
       const data = await ApiCheck.checkApi(tmpData, true);
-      tmpData.cluster_info = data.data;
+      tmpData.cluster_info = data?.data;
       const { cluster_info } = tmpData;
       // Updates the cluster-information in the registry
-      this.$scope.$emit('updateAPI', { cluster_info });
-      this.apiEntries[index].cluster_info = cluster_info;
-      this.apiEntries[index].status = 'online';
-      this.apiEntries[index].allow_run_as = data.data.allow_run_as;
+      this.state.apiEntries[index].cluster_info = cluster_info;
+      this.state.apiEntries[index].status = 'online';
+      this.state.apiEntries[index].allow_run_as = data?.data.allow_run_as;
```

Comment:
**thought**: why is the optional chaining operator required?

**nitpick**: if you add an optional chaining operator, this should be done to inner properties too.